Mathlib/Topology/Algebra/Monoid.lean

```diff
@@ -399,6 +399,103 @@ instance Submonoid.continuousMul [TopologicalSpace M] [Monoid M] [ContinuousMul
+    apply continuous_mul.tendsto_nhdsSet_nhds
+    rintro x ⟨_, _⟩
+    simp_all
+
+/-- Let `M` be a topological space with a continuous multiplication operation and a `0`.
+Let `l` be a filter on `M` which is disjoint from the cocompact filter. Then, the multiplication map
+`M × M → M` tends to zero on the filter product `l ×ˢ 𝓝 0`. -/
+theorem tendsto_mul_prod_nhds_zero_of_disjoint_cocompact {l : Filter M}
+    (hl : Disjoint l (cocompact M)) :
+    Tendsto (fun (x : M × M) ↦ x.1 * x.2) (l ×ˢ 𝓝 0) (𝓝 0) := calc
+  (l ×ˢ 𝓝 0).map fun (x : M × M) ↦ x.1 * x.2
+  _ ≤ (𝓝ˢ (Set.univ ×ˢ {0})).map fun (x : M × M) ↦ x.1 * x.2 :=
+    Filter.map_mono <| prod_nhds_le_of_disjoint_cocompact 0 hl
+  _ ≤ 𝓝 0 := by
+    apply continuous_mul.tendsto_nhdsSet_nhds
+    rintro x ⟨_, _⟩
+    simp_all
+
+/-- Let `M` be a topological space with a continuous multiplication operation and a `0`.
+Let `l` be a filter on `M × M` which is disjoint from the cocompact filter. Then, the multiplication
+map `M × M → M` tends to zero on `Filter.coprod (𝓝 0) (𝓝 0) ⊓ l`. -/
+theorem tendsto_mul_coprod_nhds_zero_inf_of_disjoint_cocompact {l : Filter (M × M)}
+    (hl : Disjoint l (cocompact (M × M))) :
+    Tendsto (fun (x : M × M) ↦ x.1 * x.2) (Filter.coprod (𝓝 0) (𝓝 0) ⊓ l) (𝓝 0) := by
+  have := calc
+    Filter.coprod (𝓝 0) (𝓝 0) ⊓ l
+    _ ≤ Filter.coprod (𝓝 0) (𝓝 0) ⊓ l.map Prod.fst ×ˢ l.map Prod.snd      :=
+      inf_le_inf_left _ le_prod_map_fst_snd
+    _ ≤ 𝓝 0 ×ˢ l.map Prod.snd ⊔ l.map Prod.fst ×ˢ 𝓝 0                    :=
+      Filter.coprod_inf_prod_le _ _ _ _
+  apply Tendsto.mono_left (h := this)
+  apply Tendsto.sup
+  · apply tendsto_mul_nhds_zero_prod_of_disjoint_cocompact
+    exact disjoint_map_cocompact continuous_snd hl
+  · apply tendsto_mul_prod_nhds_zero_of_disjoint_cocompact
+    exact disjoint_map_cocompact continuous_fst hl
+
+/-- Let `M` be a topological space with a continuous multiplication operation and a `0`.
+Let `l` be a filter on `M × M` which is both disjoint from the cocompact filter and less than or
+equal to `Filter.coprod (𝓝 0) (𝓝 0)`. Then the multiplication map `M × M → M` tends to zero on
+`l`. -/
+theorem tendsto_mul_nhds_zero_of_disjoint_cocompact {l : Filter (M × M)}
+    (hl : Disjoint l (cocompact (M × M))) (h'l : l ≤ Filter.coprod (𝓝 0) (𝓝 0)) :
+    Tendsto (fun (x : M × M) ↦ x.1 * x.2) l (𝓝 0) := by
+  simpa [inf_eq_right.mpr h'l] using tendsto_mul_coprod_nhds_zero_inf_of_disjoint_cocompact hl
+
+/-- Let `M` be a topological space with a continuous multiplication operation and a `0`.
+Let `f : α → M` and `g : α → M` be functions. If `f` tends to zero on a filter `l`
+and the image of `l` under `g` is disjoint from the cocompact filter on `M`, then
+`fun (x : α) ↦ f x * g x` also tends to zero on `l`.
+
+This is the same theorem as `Tendsto.tendsto_mul_zero_of_disjoint_cocompact'`, except the
+hypotheses on `f` and `g` are swapped. -/
+theorem Tendsto.tendsto_mul_zero_of_disjoint_cocompact {f g : α → M} {l : Filter α}
```

Comment:
What do you think about naming this one `Tendsto.mul_zero_of_disjoint_cocompact_right` and the other one `Tendsto.mul_zero_of_disjoint_cocompact_left`?